Update button variant styles

yoshiemuranaka opened the pull request:
Adding styles to leverage bootstrap-vue button components.
Using the link variant as a replacement for our current 'ghost'
button style. Since bootstrap buttons are rounded by default
and rounded vs straight corners isn't agreed on by community,
creating a global overrides file to capture any company specific
style overrides.

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

yoshiemuranaka committed:
Update button variant styles

Adding styles to leverage bootstrap-vue button components.
Using the link variant as a replacement for our current 'ghost'
button style. Since bootstrap buttons are rounded by default
and rounded vs straight corners isn't agreed on by community,
creating a global overrides file to capture any company specific
style overrides.

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

yoshiemuranaka commented:
@derick-montague thoughts on overrides file?

yoshiemuranaka commented:
The secondary button styles are based on Jandra's design proposals, which moves away from our current outlined style

derick-montague commented:
> @derick-montague thoughts on overrides file?

I was thinking about this the other day. We should create a specific overrides file and import it last. This will be a file that wouldn't be checked in upstream in the production application. It would be downstream only, which would make it easy to manage. 

I recommend creating the overrides folder and create a file for each override area. Check it in on the prototype, but then we move it downstream when we start porting the application over to production.

yoshiemuranaka committed:
Update button font weight

Adding heavier font weight for button text to match
designs.

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

derick-montague reviewed (commented):
This looks good. I am wondering about rounded corners in general. We can turn off all rounded corners by adding this at the top of the _obmc-custom.scss file.

`$enable-rounded:   false;`

derick-montague commented on src/assets/styles/_overrides.scss:
```diff
@@ -0,0 +1,5 @@
+
+.btn {
+  border-radius: 0;
+}
```
What I would suggest is to create an overrides folder with an index.scss and then create files for specific overrides.

Also, for getting this into master, I already commented using the global variable that Boostrap 4 provides.

yoshiemuranaka replied:
Oh, that's great. Didn't see the global flag, I assume it would take care of all rounded corners (input fields, modals, etc)? So I'll just set that variable to false, then remove this overrides and we can worry about downstream specific overrides later as we need?

yoshiemuranaka committed:
Disabled rounded styles

Remove _overrides file and use $enable-rounded flag
to disable rounded corners. This will take care of
all componets with rounded corners (input fields,
modals, etc).

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

derick-montague commented:
Migration to webui-vue in https://gerrit.openbmc-project.xyz/c/openbmc/webui-vue/+/28759
